lib/pact_broker/client/cli/pact_commands.rb

```diff
@@ -116,6 +116,19 @@ def branch
                 end
               end
 
+              def consumer_app_version
+                require 'pact_broker/client/git'
+                # if I pass the flag `--consumer-app-version  ` then options.consumer_app_version == 'consumer_app_version')
+                # This stops a user passing the value consumer_app_version if they provide the flag but no value. If they provide the value "", it also now fails
+                if (options.consumer_app_version.nil? || options.consumer_app_version == '' || options.consumer_app_version == 'consumer_app_version') && options.auto_detect_version_properties
```

Comment:
Add ` using PactBroker::Client::StringRefinements` underneath ` using PactBroker::Client::HashRefinements` and then you can do `options.consumer_app_version.blank?`

Reply:
nice